lib/shopify_toolkit/schema.rb

```diff
@@ -176,4 +349,43 @@ def generate_schema_content
     content.puts "end"
     content.string
   end
+
+  def execute_metaobject_definitions(schema_content)
+    # Create a filtered version that only includes metaobject definitions
+    metaobject_content = filter_schema_content(schema_content, :metaobject)
+    eval_schema_content(metaobject_content)
+  end
+
+  def execute_metafield_definitions(schema_content)
+    # Create a filtered version that only includes metafield definitions
+    metafield_content = filter_schema_content(schema_content, :metafield)
+    eval_schema_content(metafield_content)
+  end
+
+  def filter_schema_content(schema_content, type)
+    lines = schema_content.lines
+    filtered_lines = []
+
+    # Always include the header and footer
+    filtered_lines << lines.first(8) # Header lines up to "ShopifyToolkit::Schema.define do"
+    filtered_lines.flatten!
+
+    lines.each do |line|
+      case type
+      when :metaobject
+        if line.strip.start_with?("create_metaobject_definition")
+          filtered_lines << line
+        end
+      when :metafield
+        filtered_lines << line if line.strip.start_with?("create_metafield")
+      end
+    end
+
+    filtered_lines << "end\n" # Closing line
+    filtered_lines.join
+  end
+
+  def eval_schema_content(content)
+    instance_eval(content)
+  end
```

Comment:
I'm not to fond of this eval dance, but we can roll with it for now and improve later, my alternative idea would be to enforce the schema definition to declare things in the right order and keep using `load` here.

In other words if a metaobject definition comes after a metafield definition we throw an error and call it a day. We expect the schema to be generated anyway so it's acceptable to just bail out.